scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -87,24 +88,21 @@
     (api/initialize-node-watch api-client current-nodes-atom)
     (reset! pool->fenzo-atom pool->fenzo)
 
-    ; TODO(pschorf): Figure out a better way to plumb these through
-    (chime/chime-at (tp/periodic-seq (t/now) (t/seconds 2))
-                    (fn [_]
-                      (try
-                        (let [nodes @current-nodes-atom
-                              pods @current-pods-atom
-                              offers (generate-offers nodes pods this)
-                              pool (or (config/default-pool) "no-pool")
-                              chan (pool->offers-chan pool)] ; TODO(pschorf): Support pools
-                          ;(log/info "Processing offers:" offers)
-                          (scheduler/receive-offers chan match-trigger-chan this pool offers))
-                        (catch Exception e
-                          (log/error e "Exception while forwarding offers")))))
     ; TODO(pschorf): Deliver when leadership lost
     (async/chan 1))
 
   (current-leader? [this]
-    true))
+    true)
+
+  (pending-offers [this pool-name]
+    (if (or (= pool-name (config/default-pool)) ; TODO(pschorf): Support pools
+            (= pool-name "no-pool"))
+      (let [nodes @current-nodes-atom
+            pods @current-pods-atom]
+        (generate-offers nodes pods this))
```

Comment:
For that specific case, yes (since the offers are driven by node capacity, with pods removed.)

It currently doesn't support recently launched jobs, but once you merge the expected state code I'll tie that in.

Reply:
Expected state is already there. You can consume it now.